### PR TITLE
Githubapi - Add search functionality

### PR DESCRIPTION
**Pull request by farhan687**
Add search functionality

### CODE REVIEW

File: scripts/docbase.js [hunk 1]
```diff
@@ -365,64 +367,73 @@
...
+            getCommits: function() {
                 var options = Docbase.options;
                 var file_path = $route.current.params;
-                var full_path = options.github.path+'/'+file_path.version+'/'+file_path.folder+'/'+file_path.file;
-                return $http.get('https://api.github.com/repos/'+ options.github.user +'/'+ options.github.repo+'/commits?path='+full_path+'.md');
+                var full_path = options.github.path + '/' + file_path.version + '/' + file_path.folder + '/' + file_path.file;
+                return $http.get('https://api.github.com/repos/' + options.github.user + '/' + options.github.repo + '/commits?path=' + full_path + '.md');
+            },
+            searchIndex:function(){
+                return $http.get('https://raw.githubusercontent.com/appbaseio/Docs/gh-pages/search-index.json');
```
Comment (mateusfreira) [inline scripts/docbase.js:377]: This url should be a parameter. Doesn't it?

File: scripts/search-index-v2.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+[
```
Comment (mateusfreira) [inline scripts/search-index-v2.json:1]: I think this file should not be committed... or we should simplify it...

File: scripts/search-v2.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    str = data+'';
+    var hash_arr = ['\n# ','\n## ','\n### '];
+    
+    var hash_obj = [{
+      title:'sample',
+      child:[]
+    }];
+    hash_inside_arr  = create_arr(0, str, '#');
+    
+    console.log(JSON.stringify(hash_obj));
```
Comment (mateusfreira) [inline scripts/search-v2.js:16]: Console log should be removed....

File: search.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+<!doctype html>
```
Comment (mateusfreira) [inline search.html:1]: This is a debug file? i think it is not part of project...

**Comment by siddharthlatest** [comment]
@farhan687 this is a merge request from before we talked, right?

File: scripts/docbase.js (hunk 1 above)
Comment (siddharthlatest) [inline scripts/docbase.js:377]: @mateusfreira Yes, plus the search functionality here would only work for docbase. It wouldn't work for the generated html files (after grunt-docbase)

**Comment by farhan687** [comment]
Yes, then we decide to change the way, so, this pull request doesn't make any sense.

**Pull request closed without merging**